### PR TITLE
[FLINK-26405] Add validation check of num of JM replica

### PR DESCRIPTION
**Pull request by bgeng777**
- Add JM replicas check in the validator to make sure when HA is enabled, the num of JM should be 1.
- Add the relevant test.

### CODE REVIEW

**Comment by bgeng777** [comment]
cc @wangyang0918 @gyfora

**Review by gyfora — APPROVED** [review]
Looks good! 👍

File: flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/DefaultDeploymentValidator.java [hunk 1]
```diff
@@ -86,12 +87,26 @@
...
 
-        return validateResources("JobManager", jmSpec.getResource());
+        return firstPresent(
+                validateResources("JobManager", jmSpec.getResource()),
+                validateJmReplicas("JobManager", jmSpec.getReplicas(), confMap));
+    }
+
+    private Optional<String> validateJmReplicas(
+            String component, int replicas, Map<String, String> confMap) {
+        if (!confMap.containsKey(HighAvailabilityOptions.HA_MODE.key()) && replicas != 1) {
```
Comment (wangyang0918) [inline flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/DefaultDeploymentValidator.java:102]: We should use `HighAvailabilityMode.isHighAvailabilityModeActivated(conf)` instead here.

The following is the pre-check in Flink.
```
    public int getReplicas() {
        final int replicas =
                flinkConfig.get(KubernetesConfigOptions.KUBERNETES_JOBMANAGER_REPLICAS);
        if (replicas < 1) {
            throw new IllegalConfigurationException(
                    String.format(
                            "'%s' should not be configured less than one.",
                            KubernetesConfigOptions.KUBERNETES_JOBMANAGER_REPLICAS.key()));
        } else if (replicas > 1
                && !HighAvailabilityMode.isHighAvailabilityModeActivated(flinkConfig)) {
            throw new IllegalConfigurationException(
                    "High availability should be enabled when starting standby JobManagers.");
        }
        return replicas;
    }
```

Comment (gyfora) [reply flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/DefaultDeploymentValidator.java:102]: you are right @wangyang0918 that would be a bit cleaner, we could push a small fixup for this or include this in some followup work. 

Sorry for merging this quickly

Comment (bgeng777) [reply flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/DefaultDeploymentValidator.java:102]: My bad for using wrong judgement method. I have followed Yang's advice to fix it in this new #30.

**Pull request closed without merging**